src/Resistor/ResistorInput.tsx

```diff
@@ -0,0 +1,168 @@
+import * as React from 'react';
+import { ResistorColorEntry as Color, Resistor } from 'puzzle-lib';
+import { DropdownButton, MenuItem } from 'react-bootstrap';
+import './ResistorInput.css';
+
+type Props = {
+};
+
+type State = {
+  value: String,
+  bands: (Color | null)[],
+  colorTable: Color[]
+};
+
+class ResistorInput extends React.Component<Props, State> {
+
+  private static readonly NULL_COLOR = new Color('None', '#DEC0DE', 0, undefined, undefined);

```

Comment:
It's cute, but maybe `Color` should just have default values in the constructor?  Are these values use anywhere or would you be better off just using `''` instead?

Reply:
The first value (None) is used for the display name.  The rest aren't being used anywhere.  ResistorColorEntry wasn't really designed to be extended or used externally - the preset values in the Resistor class are all that REALLY exist.  This is used simply as a known constant for "not a color" for UI purposes.  This seemed cleaner than having to hardcode the none case in all of the shared functionality elsewhere.